src/query_results.js

```diff
@@ -308,45 +308,45 @@ module.exports = class QueryResult {
      *
      * There are two cases where we need to consolidate preresults:
      * 1. one or more query nodes have an 'is_set' param
-     * 2. one or more primaryID pairs have multiple kgEdges each
+     * 2. one or more primaryCurie pairs have multiple kgEdges each
      *
      * We perform consolidation by first grouping preresults by uniqueResultID and
      * then merging each of those groups into a single consolidatedPreresult.
      */
 
-    const preresultsByUniqueResultID = {};
-    preresults.forEach((preresult) => {
-      // example inputPrimaryID and outputPrimaryID in a preresult:
+    const ucResultsByResultID = {};
+    unconsolidatedResults.forEach((ucResult) => {
+      // example inputprimaryCurie and outputprimaryCurie in a preresult:
       // [
-      //   {"inputPrimaryID": "NCBIGene:3630", "outputPrimaryID", "MONDO:0005068"},
-      //   {"inputPrimaryID": "MONDO:0005068", "outputPrimaryID", "PUBCHEM.COMPOUND:43815"}
+      //   {"inputprimaryCurie": "NCBIGene:3630", "outputprimaryCurie", "MONDO:0005068"},
```

Comment:
Any reason for lowercase `p` in `primary` here?

Reply:
Also, more generally, is it clear to everyone that `inputPrimaryCurie` and `outputPrimaryCurie` refer to nodes? We specify `edge` and `node` for `QEdge` and `QNode` but not for `inputPrimaryCurie`. Maybe it's obviously because it says `input` and `output`? Not saying we should change it, but just wanted to check.

Reply:
I'll fix the lowercase. It felt relatively clear to me that input and output are nodes on an edge, however I'd like to hear @colleenXu and @marcodarko's opinions on that. Part of the goal here is to make variable names more obvious so if anyone thinks this should be made more obvious (e.g. `inputNodePrimaryCurie`) then I'm all for it.

Reply:
I'd like to hear @marcodarko 's view. 

In general, I find the use of input/output to be confusing.....
* are they nouns (nodes) or adjectives (descriptive)?
* What's the perspective?
    * Users are giving TRAPI qGraphs with directed qEdges, and those qEdges often have to be "reversed" in actual execution....which then makes it confusing to say what is "input" and "output" related to those qEdges...
    * I think apiEdges / records are a bit clearer on what is "input" and "output" (it's what ID you give the API and what concept ID you get from the API response).
    * Do we want to distinguish the two different things above?
    * There may be more ideas related to input/output that I'm not thinking of (biomedical ID resolver's step of giving an "input" to the SRI ID resolver and receiving "output"....maybe that's a thing?)

Reply:
The definition of I/O is a bit dynamic from my experience, it depends on the direction it takes based on the edge's subject/object ids.  So it's hard to come up with a single name for them.  Similarly I was confused by that part when I wanted to refer to them as subject/object instead when I first started working on the edge manager but ended up making sense when you think of it more as a definition that changes based on the direction it takes by default I --> O but can flip to O <-- I

Reply:
@marcodarko 

My questions below might not be helpful....(I'm a bit confused here)

It sounds like you're using subject/object for the qNodes based on the qEdge's direction?

and then you use I/O for the execution of sub-queries / records? where "I" corresponds to subject and "O" to object? Does "I --> O / O <-- I" refer to different ways of executing the qEdges (reversal), or to the directionality of records, or...?

Reply:
to execution, the part I'm more familiar with.  I believe however that same check happens with the query results.  Anders checks the reversal to get the I/O from the correct node.

Reply:
I just don't see a clear way to keep the naming consistent there, unless after everything is done we "reset" everything back to the original query directionality maybe??

Reply:
I feel that this may need to be a semi-separate issue to clarify in a later pass.

Reply:
Leaving this conversation unresolved for ease of reference in the future.